sensor/kubernetes/listener/resources/selectorwrapper.go

```diff
@@ -0,0 +1,85 @@
+package resources
+
+import (
+	"math"
+
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// SelectorWrapper holds a selector and information allowing for additional checks before matching
+type selectorWrapper struct {
+	selector  selector
+	numLabels uint
+	matchNil  bool
+}
+
+func (s *selectorWrapper) getSelector() selector {
+	return s.selector
+}
+
+func (s *selectorWrapper) matches(labels labels.Labels, numLabels uint) bool {
+	if s.numLabels > numLabels {
+		return false
+	}
+	if s.numLabels == 0 {
+		return s.matchNil
+	}
+	return s.selector.Matches(labels)
+}
+
+// selector is a restricted version of labels.Selector
+type selector interface {
+	Matches(labels.Labels) bool
+}
```

Comment:
I found, that there is only one place where we use this function: 

`sel.Matches(labels.Set(wrap.PodLabels))` `in sensor/kubernetes/listener/resources/deployment_store.go`

however, `wrap.PodLabels` is just a simple map, so instead modifying it to accept a Set + Length, we could create another interface that would allow us to use both inputs.

Example:

```go
type LabelsHavingLength interface {
	Has(label string) (exists bool)
	Get(label string) (value string)
	Len() int
}
```

and then change the existing interface to 

```go
type selector interface {
	Matches(labels.LabelsHavingLength) bool
}
```

(please treat `LabelsHavingLength` as a draft name)

This would allow us to make the code more compact and introduce less changes. Would you like to give it a try?

Reply:
Sure, this is similar to what Luis suggested too, with the difference that I don't need to cast the old label.labels to the new label.labelsHavingLength (draft name). I'll go with this

Reply:
You are right, but please try to avoid casting - this is only one place that uses it, so simply create the object from the map instead of casting.

Note that `LabelsHavingLength` implements `labels.Labels` so it is fully compatible.

Reply:
Actually I just realized that the label.Labels interface is our own, so in this case I can just add a Len or NumLabels function to it

Reply:
Nevermind that, it is part of kubernetes.

Reply:
That would be easy if it is ours - but still, we should not change it ;) I think it is always better to forget about good old Java and define the interface on the consumer side (even if it is a duplicate).